File: courses/C# 2008 - Impacta/C# 2008 - M1/16.MDI/prjOficina/prjOficina/VeiculoInfo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace prjOficina
{
    class VeiculoInfo
    {

        public VeiculoInfo()
        {
            caminho = @"c:\oficina.txt";
        }

        public VeiculoInfo(string CaminhoDoArquivo)
        {
            caminho = CaminhoDoArquivo;
        }

        private string _placa;
        private string _modelo;
        private string _cor;
        private short _ano;

        private string caminho;

        public string Placa
        {
            get { return _placa; }
            set { _placa = value; }
        }

        public string Modelo
        {
            get { return _modelo; }
            set { _modelo = value; }
        }

        public string Cor
        {
            get { return _cor; }
            set { _cor = value; }
        }

        public short Ano
        {
            get { return _ano; }
            set { _ano = value; }
        }

        public void Gravar()
        {
            string registro =
                string.Format("{0};{1};{2};{3};{4}", this.Placa,this.Modelo,
                        this.Cor,this.Ano.ToString(),DateTime.Now);

            System.IO.StreamWriter sw = 
                new System.IO.StreamWriter(caminho, true);

            sw.WriteLine(registro);
            sw.Close();
        }

        public void Abrir()
        {
            if (System.IO.File.Exists(caminho))
            {
                //Shell do Sistema Operacional
                //Serve para abrir arquivos, executáveis, URLs, etc...
                System.Diagnostics.Process.Start(caminho);
            }
            else
            {
                throw new Exception("Arquivo " + caminho + "não encontrado");
            }
        }
    }
}

File: courses/C# 2008 - Impacta/C# 2008 - M2/Instrutor/CSModII20090918/VB.NET2008 Mod2 Hanna/VBNet II/Módulo II/Laboratórios/Capitulo04_Laboratorio02/Oficina.Modelos/VeiculoInfo.vb
Public Class VeiculoInfo

    Private _placa As String
    Public Property Placa() As String
        Get
            Return _placa
        End Get
        Set(ByVal value As String)
            _placa = value
        End Set
    End Property


    Private _modelo As String
    Public Property Modelo() As String
        Get
            Return _modelo
        End Get
        Set(ByVal value As String)
            _modelo = value
        End Set
    End Property


    Private _cor As String
    Public Property Cor() As String
        Get
            Return _cor
        End Get
        Set(ByVal value As String)
            _cor = value
        End Set
    End Property


    Private _ano As Short
    Public Property Ano() As Short
        Get
            Return _ano
        End Get
        Set(ByVal value As Short)
            _ano = value
        End Set
    End Property

End Class
